src/client/datascience/ipywidgets/notebookIPyWidgetCoordinator.ts

```diff
@@ -36,11 +36,12 @@ export class NotebookIPyWidgetCoordinator implements INotebookKernelResolver {
     ): Promise<void> {
         // Create a handler for this notebook if we don't already have one. Since there's one of the notebookMessageCoordinator's for the
         // entire VS code session, we have a map of notebook document to message coordinator
-        let promise = this.messageCoordinators.get(document.uri.toString());
+        const promise = this.messageCoordinators.get(document.uri.toString());
```

Comment:
You can end up in a situation here where the promise is not set on the return for line 46. I suggest you set it to the CommonMessageCoordinator.create value (instead of just sticking it into the map)

Reply:
Thank you for spotting this! I completely overlooked the "create if missing" pattern here.

Reply:
This might also explain the failing test, since this promise would never have been awaited.